xilem/src/view/flex.rs

```diff
@@ -269,23 +270,110 @@ impl ElementSplice<FlexElement> for FlexSplice<'_> {
     }
 }
 
+/// A trait which extends a [`WidgetView`] with methods to provide parameters for a flex item, or being able to use it interchangably with a spacer
+pub trait FlexExt<State, Action>: WidgetView<State, Action> {
+    /// Applies [`impl Into<FlexParams>`](`FlexParams`) to this view, can be used as child of a [`Flex`] [`View`]
+    ///
+    /// # Examples
+    /// ```
+    /// use xilem::{view::{button, label, flex, CrossAxisAlignment, FlexSpacer, FlexExt}};
+    /// # use xilem::{WidgetView};
+    ///
+    /// # fn view<State: 'static>() -> impl WidgetView<State> {
+    /// flex((
+    ///     button("click me", |_| ()).flex(2.0),
+    ///     FlexSpacer::Fixed(2.0),
+    ///     label("a label").flex(CrossAxisAlignment::Fill),
+    ///     FlexSpacer::Fixed(2.0),
+    /// ))
+    /// # }
+    ///
+    /// ```
+    fn flex(self, params: impl Into<FlexParams>) -> FlexItem<Self, State, Action>
+    where
+        State: 'static,
+        Action: 'static,
+        Self: Sized,
+    {
+        flex_item(self, params)
+    }
+
+    /// Turns this [`WidgetView`] into an [`AnyFlexChild`],
+    /// which can be used interchangably with an `FlexSpacer`, as child of a [`Flex`] [`View`]
+    ///
+    /// # Examples
+    /// ```
+    /// use xilem::{view::{flex, label, FlexSpacer, FlexExt, AnyFlexChild}};
+    /// # use xilem::{WidgetView};
+    ///
+    /// # fn view<State: 'static>() -> impl WidgetView<State> {
+    /// flex([label("a label").into_any_flex(), AnyFlexChild::Spacer(FlexSpacer::Fixed(1.0))])
+    /// # }
+    ///
+    /// ```
+    fn into_any_flex(self) -> AnyFlexChild<State, Action>
+    where
+        State: 'static,
+        Action: 'static,
+        Self: Sized,
+    {
+        AnyFlexChild::Item(flex_item(self.boxed(), FlexParams::default()))
+    }
+}
+
+impl<State, Action, V: WidgetView<State, Action>> FlexExt<State, Action> for V {}
+
+/// A `WidgetView` that can be used within a [`Flex`] [`View`]
 pub struct FlexItem<V, State, Action> {
     view: V,
     params: FlexParams,
     phantom: PhantomData<fn() -> (State, Action)>,
 }
 
+/// Applies [`impl Into<FlexParams>`](`FlexParams`) to the [`View`] `V`, can be used as child of a [`Flex`] [`View`]
+///
+/// # Examples
+/// ```
+/// use xilem::view::{button, label, flex_item, flex, CrossAxisAlignment, FlexSpacer};
+/// # use xilem::{WidgetView};
+///
+/// # fn view<State: 'static>() -> impl WidgetView<State> {
+/// flex((
+///     flex_item(button("click me", |_| ()), 2.0),
+///     FlexSpacer::Fixed(2.0),
+///     flex_item(label("a label"), CrossAxisAlignment::Fill),
+///     FlexSpacer::Fixed(2.0),
+/// ))
+/// # }
+///
+/// ```
```

Comment:
Ideally we'd have an image here. We really should set up `linebender.org/assets` at some point

(To be clear, this is fine as-is)

Reply:
Yeah our documentation story can certainly be improved, but I hope this PR does improve it a little bit at least.

Reply:
I also wonder, if the doc-tests could be combined with snapshots, such that this really also tests the logic (while reducing the amount of manual tests) (But I doubt that this is simple, if at all possible within in reason)